### ISSUE TITLE
Recording notification erratically appears and stays

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#21544]
- [ ] (1) Since a few minor releases we have the following:
- Have your default profile as 'Browse map', and turn off `Trip recording plugin/Settings/Notification `
- Start the app, change to e.g. the pedestrian profile
- Select a destination and calculate a route
- **The recording notification comes on** in the Notification bar (for no good reason, did we make the notification profile dependent?)
-  Cancel the route - **notfication stays on!** May even swipe the OsmAnd app off - it stays on. You need to force-stop OsmAnd to get rid of it again.

- [ ] (2) Side finding: 
`Trip recording plugin/Settings/Notification ` seems a misnomer? The "Notification" (for the foreground service) is mandatory to show, anyway. So this setting merely controls "**Show 'Record' button in notification bar**". And deserves a corresponding renaming to avoid a frequent misunderstanding.

**Comment by Nine-Tailed** [comment]
The issue has already been raised, we users were advised to disable notifications in the osmand settings.
https://github.com/osmandapp/OsmAnd/issues/20571

**Comment by vshcherb** [comment]
Yes that's how notification supposed to work, we can disable by default or delete it

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced with the following findings:  

1. **Profile-Specific Notifications:**  
   - The **Trip recording plugin > Settings > Notification** option is tied to individual profiles.  
   - For example, disabling notifications in the **Browse map** profile does not affect the **Walking** profile.  
   - To completely disable this notification, you need to turn it off for each profile separately.  

2. **Unexpected Notification Behavior:**  
   - With the **Notification** setting enabled in the Trip recording plugin:  
     - **Steps:**  
       - Enable notifications for Trip recording.  
       - Select a destination and calculate a route.  
       - Cancel the route.  
       - Swipe the app off.  
     - **Observed Behavior:**  
       - The recording notification appears unexpectedly in the notification bar after starting navigation.  
       - It does not disappear after canceling the route.  
       - It persists even after swiping off the app 
       
       OsmAnd~ 5.0.0#3919m, released: 2024-12-10

<table style="width:100%">
  </tr>
  <tr>
    <td> <video source src="https://github.com/user-attachments/assets/e942d701-19e0-4955-9f6d-9320a25eb3c5"  width="300" height="auto"/>   </td>
    <td><video source src="https://github.com/user-attachments/assets/ef5bfe77-7532-4eb7-88af-0dd5c4c34ab8"  width="300" height="auto"/>   </td>
  </tr>
</table>

**Comment by sonora** [comment]
> Yes that's how notification supposed to work, we can disable by default or delete it

Just to be clear, two things here:
1. Our setting does NOT control the visibility of the Android notification in the System Notification bar. Seeing such a notification if and while a foreground service is running (like during a recording) is mandatory anyway.
2. Our Setting merely controls if a trip recording should be triggerable from they system notification bar (even before a trip recording has been started).
  2.1 Hence we should change the setting name from "**Notification**" to "**Show 'Record' button in notification bar**"?
  2.2 The default state for this setting should be changed to "off"?
  2.3 We should probably make this a global setting, not profile dependent?
  2.4 Some users are troubled by the fact that this remains on when they swiped off the app. But I guess this is an acceptable feature if we change the default - then if they explicitly enabled it,  it should be fine.

**Comment by vshcherb** [comment]
Changed to disable, probably it should be deleted. As it simply suggestion to record trip
